daemon/src/maker_cfd_actor.rs

```diff
@@ -307,14 +323,62 @@ impl Handler<TakeOrder> for MakerCfdActor {
         let cfd = Cfd::new(
             current_order.clone(),
             msg.quantity,
-            CfdState::Accepted {
+            CfdState::IncomingOrderRequest {
                 common: CfdStateCommon {
                     transition_timestamp: SystemTime::now(),
                 },
+                taker_id: msg.taker_id,
             },
         );
         insert_cfd(cfd, &mut conn).await?;
 
+        self.cfd_feed_actor_inbox
+            .send(load_all_cfds(&mut conn).await?)?;
+
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl Handler<AcceptOrder> for MakerCfdActor {
+    async fn handle(&mut self, msg: AcceptOrder, _ctx: &mut Context<Self>) -> Result<()> {
+        tracing::debug!(%msg.taker_id, %msg.order_id,
+                        "Taker accepts an order."
+        );
+
+        let mut conn = self.db.acquire().await?;
+
+        // Validate if order is still valid
+        // TODO: shall we also reject if the order was already requested by someone else?
+        let current_order = match self.current_order_id {
+            Some(current_order_id) if current_order_id == msg.order_id => {
+                load_order_by_id(current_order_id, &mut conn).await.unwrap()
+            }
+            _ => {
+                self.takers()?
+                    .do_send_async(maker_inc_connections_actor::TakerMessage {
+                        taker_id: msg.taker_id,
+                        command: TakerCommand::NotifyInvalidOrderId { id: msg.order_id },
+                    })
+                    .await?;
+                // TODO: Return an error here?
+                return Ok(());
+            }
+        };
+
+        // Update order in db
+        insert_new_cfd_state_by_order_id(
+            current_order.id,
+            CfdState::Accepted {
+                common: CfdStateCommon {
+                    transition_timestamp: SystemTime::now(),
+                },
+            },
+            &mut conn,
+        )
+        .await
+        .unwrap();
+
         self.takers()?
             .do_send_async(maker_inc_connections_actor::TakerMessage {
                 taker_id: msg.taker_id,
```

Comment:
Instead of `msg.taker_id`, we should be grabbing the `TakerId` from the state of the `Cfd`. This will also verify that the `Cfd` is in fact in the correct state.